lib/Sema/TypeCheckAttr.cpp

```diff
@@ -4070,13 +4040,9 @@ void AttributeChecker::visitTransposingAttr(TransposingAttr *attr) {
 
   // Check if original function type matches expected original function type
   // we computed.
-  Optional<std::function<bool(GenericSignature *, GenericSignature *)>>
-     genericComparison;
-  genericComparison = [&](GenericSignature *a, GenericSignature *b) {
-    return a == b;
-  };
-  // TODO(bartchr): remove the lambda of `checkFunctionSignature` defined in
```

Comment:
cc @bartchr808 regarding deduplicating `checkFunctionSignature` function. This is now done.